test/cql-pytest/test_describe.py

```diff
@@ -2567,3 +2568,90 @@ def aux(row: DescRow) -> DescRow:
             assert restore_stmts == res
         finally:
             cql.execute(f"DROP KEYSPACE IF EXISTS {ks}")
+
+### ===========================================================================
+
+# `varchar` is an alias for `text`. Scylla defaults to `text` and we test it here.
+# It's not very important, but it's better to have a test that detects changes
+# like that and verifies we handle that case as expected.
+def test_restore_udf_with_varchar(cql, test_keyspace):
+    fn_content = """\
+        (arg {type_name})
+        RETURNS NULL ON NULL INPUT
+        RETURNS {type_name}
+        LANGUAGE lua
+        AS $$
+        return arg
+        $$;"""
+    # Get rid of indentation in the string.
+    fn_content = textwrap.dedent(fn_content)
+
+    with new_function(cql, test_keyspace, fn_content.format(type_name="varchar")) as fn:
+        udf_name = fn
+        udf_row = cql.execute(f"DESC FUNCTION {test_keyspace}.{fn}").one()
+        udf_stmt = udf_row.create_statement
+
+        expected_stmt = f"CREATE FUNCTION {test_keyspace}.{fn}{fn_content.format(type_name="text")}"
+        assert udf_stmt == expected_stmt
+
+    try:
+        cql.execute(udf_stmt)
+    finally:
+        cql.execute(f"DROP FUNCTION IF EXISTS {test_keyspace}.{udf_name}")
+
+# Verifies that restoring a UDF with built-in types works correctly.
+def test_restore_udf_with_builtin_types(cql, test_keyspace):
+    fn_content = """\
+        (free_arg int, a list<boolean>, b set<time>, c map<varint, text>, d tuple<uuid, inet>)
+        RETURNS NULL ON NULL INPUT
+        RETURNS bigint
+        LANGUAGE lua
+        AS $$
+        return 3
+        $$;"""
+    # Get rid of indentation in the string.
+    fn_content = textwrap.dedent(fn_content)
+
+    with new_function(cql, test_keyspace, fn_content) as fn:
+        udf_name = fn
+        udf_row = cql.execute(f"DESC FUNCTION {test_keyspace}.{fn}").one()
+        udf_stmt = udf_row.create_statement
+
+        expected_stmt = f"CREATE FUNCTION {test_keyspace}.{fn}{fn_content}"
+        assert udf_stmt == expected_stmt
+    
+    try:
+        cql.execute(udf_stmt)
+    finally:
+        cql.execute(f"DROP FUNCTION IF EXISTS {test_keyspace}.{udf_name}")
+
+# Verifies that restoring a UDF with UDTs as its arguments works correctly.
+# Reproduces: scylladb/scylladb#20256.
```

Comment:
I'm curious: Was this bug specific to UDFs? Do UDT names work correctly for "DESC TABLE", "DESC MATERIALIZED VIEW" and other places (?)? Did we have tests for those places?

Reply:
I experimented with them a bit and as far as I could tell, they worked correctly. IIRC, the code responsible for describing UDTs used some other interface to access the types and that's how we got the right names and not wrapped in `frozen<>`. I can take a look if we have tests verifying that once I'm done with my current tasks

Reply:
I see test_describe.py has some elaborate code for creating random types including UDTs. All this complexity obscures (at least for me) what is already being tested with UDTs. Perhaps we're already testing them with DESC TABLE and other places, and just forgot to test DESC FUNCTION with them.